src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,64 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl<T extends Machine> implements MachineService<T> {
```

Comment:
```suggestion
public class MachineServiceImpl implements MachineService<Machine> {
```

Reply:
in the implementation we need to choose some particular type instead of `T`
and because it's machine service we will be working with machines